src/chat/index.ts

```diff
@@ -496,64 +499,56 @@ export class OpenRouterChatLanguageModel implements LanguageModelV2 {
+                });
+                textStarted = true;
+              }
               controller.enqueue({
                 type: 'text-delta',
                 delta: delta.content,
                 id: generateId(),
               });
             }
 
-            if (delta.reasoning != null) {
+            const emitReasoningChunk = (chunkText: string) => {
+              if (!reasoningStarted) {
+                controller.enqueue({
+                  type: 'reasoning-start',
+                  id: generateId(),
+                });
+                reasoningStarted = true;
+              }
               controller.enqueue({
                 type: 'reasoning-delta',
-                delta: delta.reasoning,
+                delta: chunkText,
                 id: generateId(),
               });
-            }
+            };
 
+            if (delta.reasoning != null) {
+              emitReasoningChunk(delta.reasoning);
+            }
             if (delta.reasoning_details && delta.reasoning_details.length > 0) {
               for (const detail of delta.reasoning_details) {
                 switch (detail.type) {
                   case ReasoningDetailType.Text: {
-                    if (detail.text) {
-                      controller.enqueue({
-                        type: 'reasoning-delta',
-                        delta: detail.text,
-                        id: generateId(),
-                      });
-                    }
-                    if (detail.signature) {
-                      controller.enqueue({
-                        type: 'reasoning-end',
-                        id: generateId(),
-                      });
-                    }
+                    if (detail.text) emitReasoningChunk(detail.text);
                     break;
                   }
                   case ReasoningDetailType.Encrypted: {
-                    if (detail.data) {
-                      controller.enqueue({
-                        type: 'reasoning-delta',
-                        delta: '[REDACTED]',
-                        id: generateId(),
-                      });
-                    }
+                    if (detail.data) emitReasoningChunk('[REDACTED]');
                     break;
                   }
                   case ReasoningDetailType.Summary: {
-                    if (detail.summary) {
-                      controller.enqueue({
-                        type: 'reasoning-delta',
-                        delta: detail.summary,
-                        id: generateId(),
-                      });
-                    }
-                    break;
-                  }
-                  default: {
-                    detail satisfies never;
+                    if (detail.summary) emitReasoningChunk(detail.summary);
                     break;
                   }
+                  default:
+                    // exhaustive check
+                    (detail satisfies never);
```

Comment:
Can we have some error or similar thrown/logged if we hit the `default` branch? Unless we are expecting to hit it (doesn't seem like it).